src/test/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/WorkerTest.java

```diff
@@ -1474,6 +1484,71 @@ public void testEndOfShardAfterShutdownRequest() throws Exception {
 
     }
 
+    @Test
+    public void testBuilderWithDefaultKinesisProxy() {
+        IRecordProcessorFactory recordProcessorFactory = mock(IRecordProcessorFactory.class);
+
+        Worker.Builder builder = new Worker.Builder()
+            .recordProcessorFactory(recordProcessorFactory)
+            .config(config);
+        builder.build();
+        assertNotNull(builder.getKinesisProxy());
+        assertTrue(builder.getKinesisProxy() instanceof KinesisProxy);
```

Comment:
This is not the correct way to test the proxy that is set. We need to make sure that the worker is getting the proxy that we expect it to use.